packages/shorebird_cli/lib/src/cache.dart

```diff
@@ -52,13 +52,30 @@ class Cache {
 
   void registerArtifact(CachedArtifact artifact) => _artifacts.add(artifact);
 
+  static const _maxAttempts = 3;
+
   Future<void> updateAll() async {
     for (final artifact in _artifacts) {
       if (await artifact.isUpToDate()) {
         continue;
       }
 
-      await artifact.update();
+      var attempts = 0;
```

Comment:
```suggestion
      var retryAttempts = 0;
```

Reply:
Also I believe we already depend on `package:retry` so it might make sense to use that here as well. It would clean this up a bit imo.

Reply:
@felangel we don't depend on it, at least not on the CLI, but it seems interesting, should we add the dep or wouldn't it be too much to add a dep just for that?

Reply:
I went ahead and refactored to use the package, it cleared the code so much!